google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurationPanel.java

```diff
@@ -110,15 +95,26 @@ protected void textChanged(DocumentEvent e) {
   public void resetEditorFrom(@NotNull AppEngineDeploymentConfiguration configuration) {
     promoteCheckbox.setSelected(configuration.isPromote());
     versionIdField.setText(configuration.getVersion());
-    projectSelector.setText(configuration.getCloudProjectName());
     stopPreviousVersionCheckbox.setSelected(configuration.isStopPreviousVersion());
     deployAllConfigsCheckbox.setSelected(configuration.isDeployAllConfigs());
 
     if (configuration.getEnvironment() != null) {
       environmentLabel.setText(configuration.getEnvironment().localizedLabel());
     }
 
-    refreshApplicationInfoPanel();
+    // TODO(ivanporty) add project name to configuration and then use separate project ID field.
```

Comment:
I mean here when we populate from `AppEngineDeploymentConfiguration`, we don't have project name there, just project ID. Unless we open dialog and load the list to figure out the project name, we will show project ID. So if we also keep project name and ID in `AppEngineDeploymentConfiguration`, we will show proper name in `ProjectSelector`.

I confirmed that `ProjectSelector#getText` was returning ID of the project, so deployment config using `getCloudProjectName()` is actually misleading, it's ID.

Reply:
Exactly, initially the project selector will show ID instead of a name. And then after dialog opens and load project list, it can show actual project name. That's what I thought too, it needs a separate PR since `AppEngineDeploymentConfiguration` is used in many places.